app/models/spree/gateway/adyen_credit_card.rb

```diff
@@ -19,5 +20,125 @@ def method_type
     def authorize(amount, card, gateway_options)
       ActiveMerchant::Billing::Response.new(true, "dummy authorization response")
     end
+
+    # Performs and authorization call to Adyen for the payment
+    # @raise [Spree::Core::GatewayError] if the encrypted card data is missing
+    # @raise [Spree::Core::GatewayError] if the authorize call fails
+    def authorize_new_payment payment
+      response = perform_authorization(payment)
+
+      unless response.success?
+        payment.log_entries.create!(details: response.to_yaml)
+        raise Spree::Core::GatewayError.new(
```

Comment:
Should raise a custom error instead. something that inherits from Gateway error, that way you don't have to explicitly set the error message as well.